stdlib/test/collections/test_list.mojo

```diff
@@ -759,6 +759,32 @@ def test_list_contains():
     # assert_equal(List(0,1) in y,False)
 
 
+def test_list_eq():
+    var a = List[Int](1, 2, 3)
+    var b = List[Int](1, 2, 3)
+    var c = List[Int](0, 1, 2)
+    var d = List[Int](0, 1, 2, 3)
+    assert_true(a == b)
```

Comment:
Wonder if it's worth using `assert_equal` and `assert_not_equal`? Guessing we get the lhs + rhs values propagated
```suggestion
    assert_equal(a, b)
```

Reply:
@ChristopherLR Let's use both ? if you could create a new suggestion with both ?

Reply:
```suggestion
    assert_true(a == b)
    assert_equal(a, b)
```